internal/services/servicebus/servicebus_topic_resource.go

```diff
@@ -283,6 +283,27 @@ func resourceServiceBusTopicCreateUpdate(d *pluginsdk.ResourceData, meta interfa
 		return fmt.Errorf("retrieving ServiceBus Namespace %q (Resource Group %q): %+v", id.NamespaceName, id.ResourceGroupName, err)
 	}
 
+	isPremiumNamespacePartitioned := true
+	var sku namespaces.SkuName
+	if nsModel := resp.Model; nsModel != nil {
+		sku = nsModel.Sku.Name
+	}
+
+	if sbNamespaceModel := resp.Model; sbNamespaceModel != nil {
+		if sbNamespaceModel.Properties != nil &&
+			sbNamespaceModel.Properties.PremiumMessagingPartitions != nil && *sbNamespaceModel.Properties.PremiumMessagingPartitions == 1 {
+			isPremiumNamespacePartitioned = false
+		}
+	}
```

Comment:
A few things here:
- We can condense this so we only nil check the model once
- Sku is a pointer so this should be nil checked to prevent a potential panic
- Premium messaging partitions can be 0, 1, 2 or 4, I'm assuming any value other than 0 means that the namespace is partitioned so the condition should be updated to check whether premium messaging partitions is greater than 0 and set the bool to true if it is.
```suggestion
	isPremiumNamespacePartitioned := false
	var sku namespaces.SkuName
	if model := resp.Model; model != nil {
		if model.Sku != nil {
			sku = model.Sku.Name
		}
		
		if props := model.Properties; props != nil && props.PremiumMessagingPartitions != nil && props.PremiumMessagingPartitions > 0 {
			isPremiumNamespacePartitioned = true
		}
		
	}
```

Reply:
The [provider accepts 0 as a value](https://github.com/hashicorp/terraform-provider-azurerm/blob/61f652ec33ab9a08fcafd3a9a97ca728e20f1f4e/internal/services/servicebus/servicebus_namespace_resource.go#L100-L105) for partitioning. If the API doesn't allow this then we should also update the `premium_messaging_partitions` property in the `azurerm_servicebus_namespace` resource. 

Furthermore, please still make the following changes since there are redundant nil checks and variable assignments in the original.

```suggestion
	isPremiumNamespacePartitioned := true
	var sku namespaces.SkuName
	if model := resp.Model; model != nil {
		if model.Sku != nil {
			sku = model.Sku.Name
		}
		
		if props := model.Properties; props != nil && props.PremiumMessagingPartitions != nil && props.PremiumMessagingPartitions == 1 {
			isPremiumNamespacePartitioned = false
		}
		
	}
```

Reply:
We can't set the default value to `1` as the `premiumMessagingPartitions` also applies to standard namespace which has `0` as the default value.

Updated the nil check